master/internal/api_generic_tasks.go

```diff
@@ -433,12 +458,162 @@ func (a *apiServer) KillGenericTask(
+			ctx, nil, *childTask.Config, projectID,
+		)
+		if err != nil {
+			return nil, err
+		}
+		// check if job still in registry
+		_, exists := tasklist.GroupPriorityChangeRegistry.Load(*childTask.JobID)
+		if !exists {
+			priorityChange := func(priority int) error {
+				genericTaskSpec.GenericTaskConfig.Resources.SetPriority(&priority)
+				return nil
+			}
+			if err = tasklist.GroupPriorityChangeRegistry.Add(*childTask.JobID, priorityChange); err != nil {
+				return nil, err
+			}
+		}
+		allocationString, err := a.GetAllocationFromTaskID(ctx, childTask.TaskID)
+		if err != nil {
+			return nil, err
+		}
+		allocationID := model.AllocationID(allocationString)
+		logCtx := logger.Context{
+			"job-id":    childTask.JobID,
+			"task-id":   childTask.TaskID,
+			"task-type": model.TaskTypeGeneric,
+		}
+		onAllocationExit := func(ae *task.AllocationExited) {
+			syslog := logrus.WithField("component", "genericTask").WithFields(logCtx.Fields())
+			isPaused, err := a.m.db.IsPaused(ctx, childTask.TaskID)
+			if err != nil {
+				syslog.WithError(err).Error("on allocation exit")
+			}
+			if !isPaused {
+				if err := a.m.db.CompleteGenericTask(childTask.TaskID, time.Now().UTC()); err != nil {
+					syslog.WithError(err).Error("marking generic task complete")
+				}
+				if err := tasklist.GroupPriorityChangeRegistry.Delete(*childTask.JobID); err != nil {
+					syslog.WithError(err).Error("deleting group priority change registry")
+				}
+			}
+		}
+		allocationSpecifier, err := allocationID.GetAllocationSpecifier()
+		if err != nil {
+			return nil, err
+		}
+		err = task.DefaultService.StartAllocation(
+			logCtx, sproto.AllocateRequest{
+				AllocationID:      model.AllocationID(fmt.Sprintf("%s.%d", childTask.TaskID, allocationSpecifier+1)),
+				TaskID:            childTask.TaskID,
+				JobID:             *childTask.JobID,
+				JobSubmissionTime: time.Now().UTC(),
+				RequestTime:       time.Now().UTC(),
+				IsUserVisible:     true,
+				Name:              fmt.Sprintf("Generic Task %s", childTask.TaskID),
+				SlotsNeeded:       *genericTaskSpec.GenericTaskConfig.Resources.Slots(),
+				ResourcePool:      genericTaskSpec.GenericTaskConfig.Resources.ResourcePool(),
+				FittingRequirements: sproto.FittingRequirements{
+					SingleAgent: genericTaskSpec.GenericTaskConfig.Resources.IsSingleNode(),
+				},
+				Preemptible: true,
+				Restore:     false,
+			}, a.m.db, a.m.rm, genericTaskSpec, onAllocationExit)
+		if err != nil {
+			return nil, err
+		}
+		err = a.SetResumedState(ctx, childTask.TaskID)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &apiv1.ResumeGenericTaskResponse{}, nil
+}
+
 func (a *apiServer) GetAllocationFromTaskID(ctx context.Context, taskID model.TaskID,
 ) (string, error) {
 	allocation := model.Allocation{}
 	err := db.Bun().NewSelect().Model(&allocation).
 		ColumnExpr("allocation_id").
-		Where("task_id = ?", taskID).Scan(ctx)
+		Where("task_id = ?", taskID).
+		OrderExpr("start_time DESC").Scan(ctx)
```

Comment:
why do we need an order by but not a limit 1? does bun magically not explode with multiple results in this case?

Reply:
Since tasks can be paused & unpaused multiple allocations can be given to a task. so we order by start_time to get the most recent.

Reply:
what im asking is, you are scanning into a `model.Allocation` with an order by without a limit, so why is it not `[]model.Allocation`? i might be totally missing something.

Reply:
Right, not actually sure how it was working before (I assume that it is just taking the first result). Added Limit(1)